src/c++/perf_analyzer/report_writer.cc

```diff
@@ -388,4 +407,45 @@ ReportWriter::WriteGpuMetrics(std::ostream& ofs, const Metrics& metric)
   }
 }
 
+void
+ReportWriter::WriteLlmMetrics(std::ostream& ofs)
+{
+  auto [avg_first_token_latency, avg_t2t_latency] = CalculateLlmMetrics();
+  ofs << "," << avg_first_token_latency;
+  ofs << "," << avg_t2t_latency;
+}
+
+std::tuple<double, double>
+ReportWriter::CalculateLlmMetrics()
+{
+  const std::vector<Experiment>& experiments{collector_->GetData()};
+  std::vector<double> first_token_latencies;
+  std::vector<double> t2t_latencies;
+
+  for (const auto& exp : experiments) {
+    for (const auto& req : exp.requests) {
+      for (size_t i = 0; i < req.response_times_.size(); i++) {
+        if (i <= 0) {
+          const std::chrono::duration<double, std::micro> ttft{
+              req.response_times_[i] - req.start_time_};
+          first_token_latencies.push_back(ttft.count());
+        } else {
+          const std::chrono::duration<double, std::micro> t2t{
+              req.response_times_[i] - req.response_times_[i - 1]};
+          t2t_latencies.push_back(t2t.count());
+        }
+      }
```

Comment:
Not pushing this super hard but I find moving the if out of the for loop to be easier to read:
```suggestion
      if (req.response_times_.empty() == false) {
        const std::chrono::duration<double, std::micro> ttft{
            req.response_times_.front() - req.start_time_};
        first_token_latencies.push_back(ttft.count());
      }
      for (size_t i = 1; i < req.response_times_.size(); i++) {
        const std::chrono::duration<double, std::micro> t2t{
            req.response_times_[i] - req.response_times_[i - 1]};
        t2t_latencies.push_back(t2t.count());
      }
```